reana_client/validation.py

```diff
@@ -521,6 +521,31 @@ def _validate_cwl_parameters(reana_yaml):
         )
         sys.exit(1)
 
+    def _check_dangerous_operations(workflow):
+        """Check for "baseCommand" and "arguments" in workflow.
+
+        If these keys are found, validate if they have dangerous operations.
+        """
+        dang_keys = ["baseCommand", "arguments"]
+        for k in workflow.keys():
```

Comment:
I meant we don't need to iterate over all keys, as we're only interested in `baseCommand` and `arguments`. Something like this:

```python
cmd_keys = ["baseCommand", "arguments"]
for cmd_key in cmd_keys:
    if cmd_key in workflow:
        cmd_values = (
            workflow[cmd_key]
            if isinstance(workflow[cmd_key], list)
            else [workflow[cmd_key]]
        )
        for cmd_value in cmd_values:
            _validate_dangerous_operations(str(cmd_value), step=workflow.get("id"))
```